src/acq/adc.c

```diff
@@ -648,8 +651,18 @@ static void bl_acq_adc_dma_isr(bl_acq_adc_t *adc, unsigned buffer)
 	}
 
 	for (unsigned c = 0; c < adc->config.channel_count; c++) {
-		bl_acq_channel_commit_sample(
-				adc->config.channel_index[c], sample[c]);
+		if (bl_led_src[bl_led_active].src_mask & adc->config.src_mask[c]) {
+			bl_acq_channel_commit_sample(
+					adc->config.channel_index[c], sample[c]);
+
+			if (adc->config.src_mask[c] & 0xF0) {
```

Comment:
Should this not be:

```suggestion
			if (adc->config.src_mask[c] & 0xF) {
```

Reply:
Or with the `src_is_pd` suggestion:

```suggestion
			if (adc->config.src_is_pd[c]) {
```

Reply:
+1